mobile/src/main/java/org/openhab/habdroid/ui/PreferencesActivity.kt

```diff
@@ -576,16 +564,262 @@ class PreferencesActivity : AbstractBaseActivity() {
             }
         }
 
-        override fun onAvailableConnectionChanged() {
+        override fun onActiveConnectionChanged() {
+            // no-op
+        }
+
+        override fun onPrimaryConnectionChanged() {
             updateNotificationStatusSummaries()
         }
 
-        override fun onCloudConnectionChanged(connection: CloudConnection?) {
+        override fun onActiveCloudConnectionChanged(connection: CloudConnection?) {
+            // no-op
+        }
+
+        override fun onPrimaryCloudConnectionChanged(connection: CloudConnection?) {
             updateNotificationStatusSummaries()
         }
 
         companion object {
             private const val REQUEST_CODE_RINGTONE = 1000
+        }
+    }
+
+    class ServerEditorFragment : AbstractSettingsFragment() {
+        private lateinit var config: ServerConfiguration
+        private var markAsPrimary = false
+
+        override val titleResId: Int get() = R.string.settings_edit_server
+
+        override fun onCreate(savedInstanceState: Bundle?) {
+            config = requireArguments().getParcelable("config")!!
+            super.onCreate(savedInstanceState)
+            setHasOptionsMenu(true)
+        }
+
+        override fun onCreateOptionsMenu(menu: Menu, inflater: MenuInflater) {
+            super.onCreateOptionsMenu(menu, inflater)
+            inflater.inflate(R.menu.server_editor, menu)
+            val deleteItem = menu.findItem(R.id.delete)
+            deleteItem.isVisible = prefs.getConfiguredServerIds().contains(config.id)
+        }
+
+        override fun onOptionsItemSelected(item: MenuItem): Boolean {
+            return when (item.itemId) {
+                R.id.save -> {
+                    saveAndQuit()
+                    true
+                }
+                R.id.delete -> {
+                    AlertDialog.Builder(preferenceManager.context)
```

Comment:
We should use `DialogFragment`s here (and below), since otherwise the dialog will disappear on device rotation.

Reply:
Can you have a look at my latest commit? It crashes, but I didn't figure out why.

Reply:
You can't pass callbacks like that, but need to use the parent fragment as interface. I'll fix it.

Reply:
Fixed now.